yunwoo/programmers-2/숫자의-표현.js

```diff
@@ -0,0 +1,53 @@
+function solution(n) {
+  const naturalNumberArray = Array.from({ length: n }, (_, i) => i + 1);
+  let count = 1;
+
+  for (let i = 1; i <= Math.floor(naturalNumberArray.length / 2); i++) {
+    let acc = i;
+    let cur = i;
+
+    while (n >= acc) {
+      if (n === acc) {
+        count++;
+        break;
+      }
+
+      cur++;
+      acc += cur;
+    }
+  }
+
+  return count;
+}
```

Comment:
```suggestion
function solution(n) {
  const naturalNumberArray = Array.from({ length: n }, (_, i) => i + 1);
  let result = 1;

  for (let i = 1; i <= Math.floor(n / 2); i++) {
    let acc = i;
    let cur = i;

    while (n >= acc) {
      if (n === acc) {
        result++;
        break;
      }

      cur++;
      acc += cur;
    }
  }

  return result;
}
```
- 풀이 잘봤습니다~ 🙇‍♂️
- Math.floor()에서 naturalNumberArray.length는 결국 n과 같으니 n을 사용해도 될 것 같아요 :)
- 아래 풀이는  투포인터를 이용한 풀이입니다 반복문을 1회만 사용한다는 특징이 있습니다 :) 

```js
function solution(n) {
  let [i, j] = [1, 1];
  let tempValue = 0;
  let result = 0;

  while (i <= n + 1) {
    if (tempValue < n) {
      tempValue += i++;
      continue;
    }
    if (tempValue === n) {
      result++;
    }
    tempValue -= j++;
  }

  return result;
}
```

Reply:
- 아 그렇네요 주어진 n을활용할 수 있군요!!
- 반복문 1번으로 가능하다니.. 참고하겠습니다 감사합니다🙇‍♂️